### DISCUSSION TITLE
Clarify role of `pyrightconfig.json` in resolving interpreter

### DISCUSSION THREAD

**Discussion by raulparada** [microsoft/pylance-release#5684] [Enhancement]
I spent a good few hours trying to resolve an issue using syntax introduced on python 3.12. The error was:

> Function type parameter syntax requires Python 3.12 or newer (Pylance)

I looked absolutely everywhere, e.g.:
* Conflicts pyenv <> Pylance
* VS code settings
* Trace logs from language server, Pylance, etc.
* Bug reports in this repository and in the core Python extension.

...and I was convinced that Pylance was using the wrong interpreter. In the end, it was cause a `pyrightconfig.json` on the root of my working directory had `pythonVersion` set to a lower version (3.10). I had forgotten this file even existed, and the naming of things (pyright vs pylance) certainly doesn't help to figure this out.

**I think the extension should clearly communicate where a configuration is causing a conflict like this (interpreter on 3.12, config file on 3.10)**.

Thanks!

**Comment by rchiodo** [comment]
Transfering to a discussion item for up votes.
